yt_dlp/extractor/moview.py

```diff
@@ -0,0 +1,89 @@
+                'description': 'Mengapa Candy Monster ingin mengambil permen Chloe?',
+                'thumbnail': 'https://video.jixie.media/1034/146182/146182_1280x720.jpg',
+            }
+        }, {
+            # non-drm hls
+            'url': 'https://www.moview.id/play/75/Paris-Van-Java-Episode-16',
+            'info_dict': {
+                'id': '28210',
+                'ext': 'mp4',
+                'categories': [''],
+                'duration': 2595.666667,
+                'tags': [''],
+                'display_id': 'Paris-Van-Java-Episode-16',
+                'uploader_id': 'Mo165qXUUf',
+                'thumbnail': 'https://video.jixie.media/1003/28210/28210_1280x720.jpg',
+                'description': 'md5:2a5e18d98eef9b39d7895029cac96c63',
+                'title': 'Paris Van Java Episode 16',
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        video_id = self._search_regex(
+            r'video_id\s*=\s*"(?P<video_id>[^"]+)', webpage, 'video_id')
+
+        # adapted from kompas.py
+        json_data = self._download_json(
+            'https://apidam.jixie.io/api/public/stream', display_id,
+            query={'metadata': 'full', 'video_id': video_id})['data']
+
+        formats, subtitles = [], {}
+        for stream in json_data['streams']:
+            if stream.get('type') == 'HLS':
+                if json_data.get('drm'):
+                    self.write_debug('Found drm content, trying to use non-drm link')
+                else:
+                    fmt, sub = self._extract_m3u8_formats_and_subtitles(stream.get('url'), display_id, ext='mp4')
+                    formats.extend(fmt)
+                    self._merge_subtitles(sub, target=subtitles)
+            else:
+                formats.append({
+                    'url': stream.get('url'),
+                    'width': stream.get('width'),
+                    'height': stream.get('height'),
+                    'ext': 'mp4',
+                })
+
+        self._sort_formats(formats)
+        return {
+            'id': video_id,
+            'display_id': display_id,
+            'formats': formats,
+            'subtitles': subtitles,
+            'title': json_data.get('title') or self._html_search_meta(['og:title', 'twitter:title'], webpage),
+            'description': (clean_html(traverse_obj(json_data, ('metadata', 'description')))
+                            or self._html_search_meta(['description', 'og:description', 'twitter:description'], webpage)),
+            'thumbnails': traverse_obj(json_data, ('metadata', 'thumbnails')),
+            'duration': float_or_none(traverse_obj(json_data, ('metadata', 'duration'))),
+            'tags': try_call(lambda: json_data['metadata']['keywords'].split(',')),
+            'categories': try_call(lambda: json_data['metadata']['categories'].split(',')),
```

Comment:
```suggestion
            'tags': try_call(lambda: (json_data['metadata']['keywords'] or None).split(',')),
            'categories': try_call(lambda: (json_data['metadata']['categories'] or None).split(',')),
```
should fix the `['']`

But why is this empty for both tests?

Reply:
The tag and categories is empty because the website only return `""` for this link. `['']` resolved in https://github.com/yt-dlp/yt-dlp/pull/4607/commits/442a3b59e1e8e1a72d79af8df171b94b3fe376c4